### ISSUE TITLE
Make it possible to choose the Enconding when downloading files

### ISSUE THREAD

**Issue by RyckRichards** [Tatoeba/tatoeba2#2264]
As for now, people can download files from lists, for example, in .csv. but sometimes it doesn't display some sentences correctly. Besides, many people are unfamiliar with it,

I'd suggest to add a box which the user can choose

- utf8
- unicode
- ansi

**Comment by ckjpn** [comment]
I wonder if this is really necessary.

If you are using the proper software, I don't think this would be a problem for you.

I use BBEdit on a Mac, and there is an option to open the file with various encodings, but the default way files open gets UTF8 with no problem.

Perhaps someone can recommend software to use with your operating system.

**Comment by agrodet** [comment]
You're mixing different things here. 

CSV is a **file format**. It is a delimited text file that uses a comma to separate values, hence the name Comma-Separated Values. Therefore, calling CSV a file using tabulation as a separator (what we do on Tatoeba) is erroneous. TSV, Tab-separated values, is the correct way to call such a format. However, since most of our contemporaries are IT-uneducated, I guess that's a difficult task to distribute TSV files. But that's a different issue.  

Unicode is **not** an encoding. It is a **standard**. Simply said, it refers to the abstract set of characters in itself and not to any particular encoding. Therefore, no tool can propose a "Unicode" encoding, although some incorrect documentation refers to it as such.

Left from your list are UTF-8 and ANSI. From these two, ANSI is a character set. There is no such thing as **one** ANSI encoding. Most of the time, the misleading term "ANSI" refers to "whatever my Windows system uses by default", which is often Windows-1252, and sometimes refers to ASCII or its extended version encompassed in ANSI since ASCII is 7-bit where ANSI is 8-bit. Note that ANSI is western-centered, another reason to let it fall into oblivion.

Long story short, from you little list, only UTF-8 is an encoding. And actually, it is the only encoding we should share and should encourage people to use. Other encodings are old, very prompt to incompatibility between files **and systems**, and cause of broken keyboards and hair torn off all around the world :) 

Therefore I think it is a really bad idea to propose anything else that UTF. If 8 is not appropriate (I don't think so), other encodings can be considered. The fact that your list mixed four complete different things somehow support my point. 

Finally, all modern editors can open UTF-8 encoded files. You said that "sometimes it doesn't display some sentences correctly." I'm curious to see what sentences are garbled, and under which conditions.

**Comment by RyckRichards** [comment]
> Finally, all modern editors can open UTF-8 encoded files. You said that "sometimes it doesn't display some sentences correctly." I'm curious to see what sentences are garbled, and under which conditions.

download the list 9903 - https://tatoeba.org/eng/sentences_lists/download/9903
id - not checked
translation - none

file downloaded as List ptbr 01 - 2020-04-09

opening... 

Result - https://prnt.sc/rw5ldh

**Comment by alanfgh** [comment]
Ricardo, your Excel is loading the file with encoding set to Windows-1252. If you want to load it with the correct encoding, UTF-8, you have several options:

(1) Use OpenOffice.org's "Calc" (spreadsheet) program. When it opens a file with a .csv extension that contains non-ASCII text data, it gives you a choice as to which encoding to use to read it, as well as which character (comma, tab, etc.) is the delimiter. You're going to want to choose "UTF-8" and "tab".

![open_office_calc](https://user-images.githubusercontent.com/1213027/78952717-78e94480-7aa4-11ea-8ebf-69c8ed712b05.png)

(2) If you want to continue to use Excel, you can do the following: Rather than clicking on a .csv file to open it, or using "Open" from Excel, make sure you start up Excel first, then click on "Data", and finally "From Text/CSV". (It looks like you're using a different version of Excel from mine, but presumably, it also has this functionality.) When you do that, you will get a dialog that gives you a choice as to which encoding and delimiter you should use.

![excel_data_tab](https://user-images.githubusercontent.com/1213027/78952727-80a8e900-7aa4-11ea-8b1b-f1e4fa226f6e.png)

(3) Another option that you can choose with Excel is to use the legacy text wizard:

![marked_up_legacy_data_import_wizards](https://user-images.githubusercontent.com/1213027/78952859-e4cbad00-7aa4-11ea-9674-e8acaa866bd3.jpg)

But option 3 doesn't always work for me.

**Comment by alanfgh** [comment]
Regarding @agrodet's remarks, I agree with everything he said, and I have one more thing to add.

> However, since most of our contemporaries are IT-uneducated, I guess that's a difficult task to distribute TSV files. But that's a different issue.

Apart from **people**, most **programs** don't know by default what a file with a .tsv extension is. If you're sophisticated enough, you can tell your operating system to always open .tsv files with OpenOffice Calc, and it will treat it as a delimited file (and will bring up the import wizard to give you a chance to specify how it should be opened). But not everyone will know how to do that.

**Comment by RyckRichards** [comment]
Besides, many people use MS Office instead of Libre Office (me included) - either because they're not familiar or even never hear about that.

**Comment by ckjpn** [comment]
Google it:

how to open utf-8 in excel

https://www.google.com/search?q=how+to+open+utf-8+in+excel

On my computer, Excel opens in utf-8 by default.

**Comment by jiru** [comment]
About CSV files not opening correctly in Excel, there is not much we can do here except changing the format. The file CSV produced by the list download page already contains a [BOM](https://en.wikipedia.org/wiki/Byte_order_mark) to indicate the encoding (but not CSV listed in the Downloads page though). I tested opening a list downloaded as CSV on a Windows 10 machine: Excel opens the file correctly without having to explain how to open it. So I think you should update Excel or use OpenOffice/LibreOffice like Alan suggested.

Anyway, #2257 should solve this issue because it sets the list download default format to raw text file (.txt). Power users will still be able to download a TSV if they want by selecting a different format.
